service/history/historyEngine.go

```diff
@@ -1140,6 +1149,8 @@ Update_History_Loop:
 
 		// Inform timer about the new ones.
 		e.timerProcessor.NotifyNewTimer(timerTasks)
+		e.timerProcessor.NotifyNewTimer(continueAsNewTimerTasks)
```

Comment:
It seems weird we call NotifyNewTimer twice.  Can we merge it into a single call to Notify?